react/features/face-landmarks/FaceLandmarksDetector.ts

```diff
@@ -107,8 +111,19 @@ class FaceLandmarksDetector {
         workerUrl = window.URL.createObjectURL(workerBlob);
         this.worker = new Worker(workerUrl, { name: 'Face Landmarks Worker' });
         this.worker.onmessage = ({ data }: MessageEvent<any>) => {
-            const { faceExpression, faceBox, faceCount } = data;
+            const { faceExpression, faceBox, faceCount, raisedHand } = data;
             const messageTimestamp = Date.now();
+            const localParticipant = getLocalParticipant(getState());
+
+            if (!hasRaisedHand(localParticipant) && raisedHand) {
+                if (this.lastRaisedHandTimestamp
+                    && messageTimestamp - this.lastRaisedHandTimestamp >= 2000) {
+                    dispatch(raiseHand(true));
+                    this.lastRaisedHandTimestamp = null;
+                } else if (!this.lastRaisedHandTimestamp) {
```

Comment:
I think if you test this first and then unconditionally check for the interval it will make the code more readable.